src/VisualStudio/Core/Impl/Options/Converters/NullableBoolOptionConverter.cs

```diff
@@ -20,20 +17,13 @@ public NullableBoolOptionConverter(Func<bool> onNullValue)
             _onNullValue = onNullValue;
         }
 
-        public object? Convert(object value, Type targetType, object parameter, CultureInfo culture)
-        {
-            if (value is not null or bool)
-            {
-                return null;
-            }
-
-            if (value is null)
+        public object Convert(object? value, Type targetType, object parameter, CultureInfo culture)
+            => value switch
             {
-                return _onNullValue();
-            }
-
-            return (bool)value;
-        }
+                null => _onNullValue(),
+                bool b => b,
+                _ => DependencyProperty.UnsetValue
```

Comment:
Can we actually but the third case?

Reply:
Not sure I understand the question? This would be hit if the converter was used on an invalid type. In current usage it can't, but this is the right way to handle this according to documentation: https://github.com/dotnet/roslyn/pull/50620#pullrequestreview-571786219

Reply:
That answered things for me :)